Source/Core/Core/HW/WiimoteEmu/Extension/TaTaCon.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Core/HW/WiimoteEmu/Extension/Extension.h"
+
+namespace ControllerEmu
+{
+class Buttons;
+}  // namespace ControllerEmu
+
+namespace WiimoteEmu
+{
+enum class TaTaConGroup
+{
+  Center,
+  Rim,
+};
+
+class TaTaCon : public Extension3rdParty
+{
+public:
+  struct DataFormat
+  {
+    std::array<u8, 5> nothing;
```

Comment:
"Nothing" to see here? Sounds odd that they'd use 6 bytes of space and declare the _first_ 5 as nothing.

Reply:
They use 6 bytes (like almost every extension) because that's how many bytes are read from extensions in the most commonly used reporting mode. Many (all?) extensions seem to use the last byte for "button bits" and stick things like analog values in the other bytes. Nintendo might provide recommendations on data formats. It's relevant because the MotionPlus blindly mucks up bits in passthrough modes and it would clobber important values if they were in bad places.

I just implemented the layout according to this: https://wiibrew.org/wiki/Wiimote/Extension_Controllers/TaTaCon and It works in game. :p